src/rules/noUnnecessaryParensRule.ts

```diff
@@ -0,0 +1,205 @@
+                    "LiteralExpression",
+                ],
+                asChildOf: [
+                    "VariableDeclaration.initializer",
+                    "ParenthesizedExpression.expression",
+                    "CallExpression.arguments",
+                    "ExpressionStatement.expression",
+                    "*.type",
+                ],
+            }],
+            [{ default: true }],
+        ],
+        type: "typescript",
+        typescriptOnly: true,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING_FACTORY(expressionTypeName: string) {
+        return `Don't include unnecessary parentheses around ${expressionTypeName}`;
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as Options);
+    }
+}
+
+function isNodeOfKind(node: ts.Node, kindName: string) {
+    switch (kindName) {
+        case "LiteralExpression":
+            return isLiteralExpression(node);
+        case "Keyword":
+            return node.kind > ts.SyntaxKind.FirstKeyword && node.kind < ts.SyntaxKind.LastKeyword;
+        case "*":
+            return true;
+        default:
+            return node.kind === syntaxKindMapping[kindName];
+    }
+}
+
+function isParenthesizedType(node: ts.Node): node is ts.ParenthesizedTypeNode {
+    return node.kind === ts.SyntaxKind.ParenthesizedType;
+}
+
+function walk(ctx: Lint.WalkContext<Options>) {
+    if (ctx.options.withChild == undefined) {
+        ctx.options.withChild = [];
+    }
+    if (ctx.options.asChildOf == undefined) {
+        ctx.options.asChildOf = [];
+    }
+    const withChild = ctx.options.default ? [
+        ...ctx.options.withChild,
+        // ex: (foo).bar()
+        "Identifier",
+        // ex: ("abc") + "def"
+        "LiteralExpression",
+        // ex: let x: ('a') = 'a';
+        "LiteralType",
+        // ex: let x: (string) = 'b'; let x = (undefined);
+        "Keyword",
+        // ex: (new Foo())
+        "NewExpression",
+        // ex (options[0]).foo
+        "ElementAccessExpression",
+        // ex (x.a).b
+        "PropertyAccessExpression",
+        // ex (f());
+        "CallExpression",
+    ] : ctx.options.withChild;
+    const asChildOf = ctx.options.default ? [
+        // ex: let x = (1 + foo());
+        "VariableDeclaration.initializer",
+        // ex: type x = (string|number);
+        "TypeAliasDeclaration.type",
+        // ex: ((1 + 1)) + 2
+        "ParenthesizedExpression.expression",
+        // ex: foo((a), b); new Foo((a));
+        "CallExpression.arguments",
+        "NewExpression.arguments",
+        // ex: Foo<(a|b), c>; foo<(a)>();
+        "*.typeArguments",
+        // ex: (foo.bar());
+        "ExpressionStatement.expression",
+        // ex: let x: (string|number) = 3;
+        "VariableDeclaration.type",
+        // ex: function(foo: (number|string)) {}
+        "Parameter.type",
```

Comment:
I'm still in favor of using `"*.type"`. I can't come up with a case where parens are desired or even required.

Reply:
Done.